### PR TITLE
Fix Issue #106: Clipboard flag ignored if path is qualified to file

### PR DESCRIPTION
**Pull request by CheerfulPianissimo**

### CODE REVIEW

**Comment by Shinyzenith** [comment]
Hi, please rebase your changes onto freeze-feat-andreas as there's a merge conflict.

**Comment by Shinyzenith** [comment]
Hi it seems you included a commit which already exists in freeze branch. Why?

File: wayshot/src/wayshot.rs [hunk 1]
```diff
@@ -123,40 +124,42 @@ fn main() -> Result<()> {
         }
     };
 
+    let mut buffer = Cursor::new(Vec::new());
```
Comment (Shinyzenith) [inline wayshot/src/wayshot.rs:127]: Lets only create the buffer if clipboard is specified. Else this leads to an extra allocation which we can skip. Lets just call image.save here and create the buffer in the clipboard conditional block.

**Review by Shinyzenith — CHANGES_REQUESTED** [review]

File: wayshot/src/wayshot.rs (hunk 1 above)
Comment (CheerfulPianissimo) [reply wayshot/src/wayshot.rs:130]: I considered that but the stdout block also uses the buffer and I didn't want it to do the buffer write twice in the case of --clipboard + stdout. Should I create the buffer there too?

Comment (Shinyzenith) [reply wayshot/src/wayshot.rs:130]: Maybe a function can help

File: wayshot/src/wayshot.rs [hunk 2]
```diff
@@ -128,41 +129,52 @@ fn main() -> Result<()> {
...
-                        MimeType::Autodetect,
-                    )?;
-                }
-            }
+            clipboard_daemonize(buffer)?;
         }
     }
 
+    if !stdout_print && cli.clipboard {
+        let mut buffer = Cursor::new(Vec::new());
```
Comment (Shinyzenith) [inline wayshot/src/wayshot.rs:144]: Do we need the cli.clipboard conditional in the stdout conditional? Is this block outside of the conditional not enough?

```rs

let image_buf: Option<Cursor<Vec<...> = None;
if Some(path) = file {
    /// save
} else {
    /// to stdout

    image_buf = ...;
}

if cli.clipboard && image_buf = Some(image_buffer) {
    //buf stuff
}
```

**Review by Shinyzenith — CHANGES_REQUESTED** [review]

**Review by Shinyzenith — CHANGES_REQUESTED** [review]

File: wayshot/src/wayshot.rs [hunk 3]
```diff
@@ -135,14 +136,18 @@ fn main() -> Result<()> {
...
-        }
+        image_buf = Some(buffer);
     }
 
-    if !stdout_print && cli.clipboard {
-        let mut buffer = Cursor::new(Vec::new());
-        image_buffer.write_to(&mut buffer, requested_encoding)?;
+    if cli.clipboard {
+        let buffer = match image_buf {
+            Some(buf) => buf,
```
[the remark this answers is not in the document]
Comment (CheerfulPianissimo) [reply wayshot/src/wayshot.rs:144]: Done!

**Pull request merged**